tests/SharedTests/TestLeads.cs

```diff
@@ -0,0 +1,620 @@
+            }
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestCreateOpportunityOpportunityCustomerIdWrongLogicalNameFails()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+
+            lead.Id = orgAdminUIService.Create(lead);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                CreateOpportunity = true,
+                OpportunityCustomerId = lead.ToEntityReference(),
+                LeadId = lead.ToEntityReference()
+            };
+            try
+            {
+                orgAdminUIService.Execute(request);
+                Assert.Fail();
+            }
+            catch (Exception e)
+            {
+                Assert.IsInstanceOfType(e, typeof(FaultException));
+            }
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestCreateOpportunityOpportunityCustomerIdDoesNotExistFails()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+
+            lead.Id = orgAdminUIService.Create(lead);
+
+            var contact = new Contact
+            {
+                FirstName = "Test",
+                LastName = "Lead",
+                Id = Guid.NewGuid()
+            };
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                CreateOpportunity = true,
+                OpportunityCustomerId = contact.ToEntityReference(),
+                LeadId = lead.ToEntityReference()
+            };
+            try
+            {
+                orgAdminUIService.Execute(request);
+                Assert.Fail();
+            }
+            catch (Exception e)
+            {
+                Assert.IsInstanceOfType(e, typeof(FaultException));
+            }
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestSuccess()
+        {
+            var lead = new Lead
+            {
+                FirstName = "Test",
+                LastName = "Lead"
+            };
+
+            lead.Id = orgAdminUIService.Create(lead);
+
+            var request = new QualifyLeadRequest
+            {
+                Status = new OptionSetValue((int)Lead_StatusCode.Qualified),
+                LeadId = lead.ToEntityReference()
+            };
+
+            var response = orgAdminUIService.Execute(request) as QualifyLeadResponse;
+            Assert.IsNotNull(response);
+            var createdEntities = response.CreatedEntities;
+            Assert.IsNotNull(createdEntities);
+            Assert.IsTrue(createdEntities.Count == 0);
+        }
+
+        [TestMethod]
+        public void TestQualifyLeadRequestOpportunityCurrencyIdSuccess()
```

Comment:
Missing assert that lead has been qualified. This test should also fails as you need to also set CreateOpportunity if you want to set OpportunityCurrencyId